### PR TITLE
Issue 19098 saml logout

### PR DESCRIPTION
**Pull request by jdotcms**
Adding changes to support new logout strategy and saml logout in addition

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/auth/providers/saml/v1/DotSamlResource.java [hunk 1]
```diff
@@ -301,7 +301,7 @@ public void logout(@PathParam("idpConfigId") final String idpConfigId,
 					Logger.debug(this, () -> "Processing saml logout request for idpConfig id: " + idpConfigId);
 					final String logoutPath = this.samlConfigurationService.getConfigAsString(identityProviderConfiguration,
 							SamlName.DOT_SAML_LOGOUT_SERVICE_ENDPOINT_URL,
-							()->buildBaseUrlFromRequest(httpServletRequest) + "/");
+							()-> "/dotAdmin/#/public/logout");
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotcms/auth/providers/saml/v1/DotSamlResource.java:304]: do we want to cache bust here?  I think so, e.g.
`"/dotAdmin/?r=" + random + "#/public/logout"`

File: dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            final User user = PortalUtil.getUser(request);
+
+            APILocator.getLoginServiceAPI().doActionLogout(request, response);
+
+            if(null != user){
+                SecurityLogger.logInfo(this.getClass(), "User " +
+                        user.getFullName() + " (" + user.getUserId() + ") has logged out from IP: " + request.getRemoteAddr());
+            }
+
+            response.sendRedirect(Config.getStringProperty("logout.url", "/dotAdmin/#/public/logout"));
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java:50]: cache bust again 
`"/dotAdmin/?r=" + random + "#/public/logout"`

File: dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            Logger.info(this, ()-> "Logout DONE");
+        }  catch (Exception e) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java:52]: Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6263950)

File: dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+import com.dotcms.filters.interceptor.Result;
+import com.dotcms.filters.interceptor.WebInterceptor;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.SecurityLogger;
+import com.liferay.portal.model.User;
+import com.liferay.portal.util.PortalUtil;
+import com.liferay.util.StringPool;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java:11]: Codacy found an issue: [Avoid unused imports such as 'com.liferay.util.StringPool'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6263950)

File: dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java:52]: Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6263950)

File: dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java:11]: Codacy found an issue: [Avoid unused imports such as 'com.liferay.util.StringPool'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6263950)

File: dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java:11]: Codacy found an issue: [Avoid unused imports such as 'com.liferay.util.StringPool'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6263950)

File: dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+                            user.getFullName() + " (" + user.getUserId() + ") has logged out from IP: " + request.getRemoteAddr());
+                }
+            }
+
+            //response.sendRedirect(Config.getStringProperty("logout.url", "/dotAdmin/#/public/logout"));
+            final RequestDispatcher dispatcher = Config.CONTEXT
+                    .getRequestDispatcher("/html/portal/logout.jsp");
+            dispatcher.forward(request, response);
+            Logger.info(this, ()-> "Logout DONE");
+        }  catch (Exception e) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java:59]: Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6263950)

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java:11]: Codacy found an issue: [Avoid unused imports such as 'com.liferay.util.StringPool'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6263950)

File: dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+                            user.getFullName() + " (" + user.getUserId() + ") has logged out from IP: " + request.getRemoteAddr());
+                }
+            }
+
+            //response.sendRedirect(Config.getStringProperty("logout.url", "/dotAdmin/#/public/logout"));
+            final RequestDispatcher dispatcher = Config.CONTEXT
+                    .getRequestDispatcher("/html/portal/show-logout.jsp");
+            dispatcher.forward(request, response);
+            Logger.info(this, ()-> "Logout DONE");
+        }  catch (Exception e) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java:59]: Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6263950)

File: dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java:11]: Codacy found an issue: [Avoid unused imports such as 'com.liferay.util.StringPool'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6263950)

File: dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java:59]: Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6263950)

File: dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java:11]: Codacy found an issue: [Avoid unused imports such as 'com.liferay.util.StringPool'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6263950)

File: dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/cms/login/LogoutWebInterceptor.java:59]: Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6263950)

**Pull request merged**